### PR TITLE
DEV-14904: Back HDInsight HBase with ADLS Gen2

### PR DESCRIPTION
**Pull request by schwartzmanb**

### CODE REVIEW

**Comment by tamr-teamcity** [comment]
This PR doesn't appear to be linked to a DevOps/SRE jira ticket

File: examples/minimal/main.tf [hunk 1]
```diff
@@ -10,6 +10,37 @@ resource "azurerm_virtual_network" "hdinsight-vnet" {
...
+  name                = "hbasegen2principal"
+  resource_group_name = azurerm_resource_group.hdinsight-rg.name
+  location            = azurerm_resource_group.hdinsight-rg.location
+}
+
+resource "azurerm_role_assignment" "blob-owner-role-assigment" {
+  scope                = module.adls_gen2.storage_account_id
+  role_definition_name = "Storage Blob Data Owner"
+  principal_id         = azurerm_user_assigned_identity.adls_gen2_identity.principal_id
+}
```
Comment (souza-dan) [inline examples/minimal/main.tf:42]: Should we get these into a module? Even if it's a nested optional module?

**Review by souza-dan — APPROVED** [review]
LGTM

**Review by aditimarkale — APPROVED** [review]

File: examples/minimal/main.tf [hunk 2]
```diff
@@ -10,6 +10,37 @@ resource "azurerm_virtual_network" "hdinsight-vnet" {
   resource_group_name = azurerm_resource_group.hdinsight-rg.name
 }
 
+# Azure storage account and ADLS Gen2
+module "adls_gen2" {
+  source = "git::https://github.com/Datatamer/terraform-azure-adls-gen2.git?ref=1.0.0"
+
+  instance_name       = "tamrgen2hbase"
```
[the remark this answers is not in the document]
Comment (schwartzmanb) [reply examples/minimal/main.tf]: Yes, I changed our example to be less probable for someone to actually want to use

**Review by souza-dan — APPROVED** [review]
LGTM, thanks for creating that new nested module

**Pull request merged**